File: src/main/kotlin/third/F.py
dictionary = input().split(" ")
dictionary2 = {}
for elem in dictionary:
    dictionary2[elem] = elem
for word in input().split(" "):
    tempRes = word
    for i in range(len(word)):
        index = word[0:i+1]
        existed = dictionary2.get(index)
        if existed:
            tempRes = existed
            break
    print(tempRes, end=" ")

File: src/main/kotlin/third/F.kt
package third

fun main(args: Array<String>) {
    val dictionary = readln().split(" ")
    val dictionary2 = mutableMapOf<String, String>()

    for (elem in dictionary) {
        dictionary2[elem] = elem
    }

    readln().split(" ").forEach { word ->
        var tempRes = word
        for (i in 0..word.length - 1) {
            val index = word.substring(0..i)
            val existed = dictionary2[index]
            if (existed != null) {
                tempRes = existed
                break
            }
        }

        print(tempRes)
        print(" ")
    }
}